lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -1,38 +1,85 @@
 import { createElement } from 'preact';
 import classNames from 'classnames';
-import propTypes from 'prop-types';
 import { useContext, useEffect, useState, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
 
 import { Config } from '../config';
-import { submitGrade } from '../utils/grader-service';
-import { formatToNumber, validateGrade } from '../utils/validation';
 import ErrorDialog from './ErrorDialog';
 import Spinner from './Spinner';
 import SvgIcon from './SvgIcon';
+import { fetchGrade, submitGrade } from '../utils/grader-service';
 import { trustMarkup } from '../utils/trusted';
+import { formatToNumber, validateGrade } from '../utils/validation';
 import ValidationMessage from './ValidationMessage';
 
+// Grades are always stored as a value between [0-1] on the service layer.
+// Scale the grade to [0-10] when loading it into the UI and scale it
+// back to [0-1] when saving it to the api service.
+const GRADE_MULTIPLIER = 10;
+
+/**
+ * Custom useEffect function that handles fetching a student's
+ * grade and returning the result of that grade and the loading
+ * and error status of that fetch request.
+ */
+const useFetchGrade = student => {
+  const { authToken } = useContext(Config);
+  const [grade, setGrade] = useState('');
+  const [gradeLoading, setGradeLoading] = useState(false);
+  const [gradeError, setGradeError] = useState(false);
+
+  useEffect(() => {
+    if (Object.entries(student).length) {
+      // Don't bother fetching if the student object is empty
+      // Fetch the grade from the service api
+      // See https://www.robinwieruch.de/react-hooks-fetch-data for async in useEffect
+      const fetchData = async () => {
+        setGradeLoading(true);
+        setGradeError(false);
+        setGrade(''); // Clear previous grade so we don't show the wrong grade with the new student
+
+        try {
+          const response = await fetchGrade({ student, authToken });
+          setGrade(response.currentScore * GRADE_MULTIPLIER);
+        } catch (e) {
+          setGradeError(e.errorMessage ? e.errorMessage : 'Unknown error');
+        }
+        setGradeLoading(false);
+      };
+      fetchData();
+    }
+  }, [student, authToken]);
+  return [{ grade, gradeLoading, gradeError }];
```

Comment:
useFetchGrade returns `gradeError` as a matter of completeness , but I don't end up using it to keep things a bit more simple in the main component. we could remove it here too, but I wanted to get feedback about this. 

Not using it means if there is ever a error while making the GET request, we don't really show anything in the UI. I think this might be okay because GET errors would be really rare in the first place.